content/docs/reference-glossary.md

```diff
@@ -0,0 +1,98 @@
+---
+id: glossary
+title: Glossary of React Terms
+layout: docs
+category: Reference
+permalink: docs/glossary.html
+
+---
+
+## Single Page Application
+A single page application is an application that loads a single HTML page and all the necessary assets (CSS, Images & JavaScript) required for the application to run. Any interactions with the page or subsequent pages do not require a round trip to the server which means the page is not reloaded.
+
+
+## ES6/ES2015
+ ES6 AKA ES2015 is the current version of the ECMAScript Language Specification standard, which the JavaScript language is an implementation of. It includes many additions to the previous versions such as: arrow functions, classes, template literals, and `let` and `const` statements. 
+
+## Compilers
```

Comment:
Maybe mention "transpiler" as another term for this section

Reply:
I think we decided that we were going to avoid using the word transpiler because it causes confusion. See comment here from @gaearon: https://github.com/facebook/react/pull/8199#discussion_r98238151